packages/cos-setup/cos-setup

```diff
@@ -27,4 +28,9 @@ for x in "$@"; do
         yip -s "$STAGE".after "${x#cos.setup=}"
         ;;
     esac
+done
+
+# Read cmdline from dotnotation and execute yip file:
+for s in "$STAGE".before "$STAGE" "$STAGE".after; do
+    echo $CMDLINE | yip -d -s "$s"
```

Comment:
I don't really understand it, what's the expected content of CMDLINE at this point? how yip behaves with this sort of pipe setup? I mean, is yip expecting a path on stdin? I'd expect something like `cos.setup=/oem` to set the path on the command line, but thus was already covered, isn't it?

Reply:
yes, indeed this is another use case which is not very well clear from the PR, my bad!

This is the scenario:

From grub, you can edit the boot cmdline and add e.g. `stages.network[0].authorized_keys.root[0]=github:mudler` to add that github key to the root user, or set whatever we expose in the cloud-init config


The implementation side is a bit tricky, `yip` composes a yaml file from a sequence of dotnotations, e.g.: `stages.network[0].authorized_keys.root[0]=github:mudler`, is equivalent to: 
```yaml
stages:
  boot:
  - authorized_keys:
     root:
     - github:name
```
and then it parses it as a cloud-init. In this case, we read the whole cmdline, and we pipe it to `yip`, enabling the dotnotation syntax ( with `-d` ). The cmdline arguments become all a yaml then which is parsed in search for cloud-init keywords.